wicker/core/column_files.py

```diff
@@ -178,16 +183,49 @@ def _write_column(self, column_name: str) -> None:
         self.write_buffers[column_name] = self._get_new_buffer()
 
 
-class ColumnBytesFileCache:
+class ColumnBytesFileReader(abc.ABC):
+    """A reader class for column bytes files to inherit upon for caches and cacheless"""
+
+    def __init__(
+        self,
+        column_bytes_root_path: str,
+    ) -> None:
+        super().__init__()
+        self._column_bytes_root_path = column_bytes_root_path
+
+    def read(self, column_bytes_file_info: ColumnBytesFileLocationV1) -> bytes:
+        """Read data from column bytes file.
+
+        :param column_bytes_file_info: Location object to column bytes file on data store
+        :type column_bytes_file_info: ColumnBytesFileLocationV1
+        :return: bytes read from the column bytes file
+        """
+        column_concatenated_bytes_file_path = os.path.join(
+            self._column_bytes_root_path, str(column_bytes_file_info.file_id)
+        )
+
+        with open(column_concatenated_bytes_file_path, "rb") as f:
+            f.seek(column_bytes_file_info.byte_offset)
+            return f.read(column_bytes_file_info.data_size)
+
+    def resolve_pointers(
+        self,
+        example: validation.AvroRecord,
+        schema: schema.DatasetSchema,
+    ):
+        visitor = ResolvePointersVisitor(self, example, schema)
+        return visitor.resolve_pointers()
+
+
+class ColumnBytesFileCache(ColumnBytesFileReader):
     """A read-through caching abstraction for accessing ColumnBytesFiles"""
 
     def __init__(
         self,
+        column_root_path: str,
         local_cache_path_prefix: str = "/tmp",
         filelock_timeout_seconds: int = -1,
-        path_factory: S3PathFactory = S3PathFactory(),
-        storage: Optional[S3DataStorage] = None,
-        dataset_name: str = None,
+        storage: Optional[AbstractDataStorage] = None,
     ):
         """Initializes a ColumnBytesFileCache
```

Comment:
No longer applicable?

Reply:
@pickles-bread-and-butter if you don't mind, let's push the `path_factory` and `dataset_name` back into the `ColumnBytesFileCache` constructor (and pass them to the base class `__init__` also). I think the main idea here is that the `ColumnBytesFileReader` and its subclasses should be symmetric to the `ColumnBytesFileWriter` class that does take these parameters. Apologies ahead of time if this makes writing unit tests for the reader and cache classes more difficult.

Reply:
Would you be okay with at minimum retyping them? I did the retypes here but if you don't agree with that let me know and I'll change it.

The thing I dislike the most about this is that all of that information required for these is already baked into the schema and imo there should only be one source of truth. I'd probably prefer to re-write the other constructor to make that symmetric. Regardless I made the change here.

Reply:
Tests didn't end up being too bad but let me know if you don't like how they're written now.

Reply:
@pickles-bread-and-butter I think the `ColumnBytesFileCache` still isn't quite exactly what I was hoping for, but let me say quickly that retyping the parameters (to use the more general base class types) should be fine.

I think the ``__init__` should look like this:
```
class ColumnBytesFileCache:
    def __init__(
        self,
        dataset_name: Optional[str] = None,
        filelock_timeout_seconds: int = -1,
        local_cache_path_prefix: str = "/tmp",
        path_factory: WickerPathFactory = S3PathFactory(),
        storage: Optional[AbstractDataStorage] = None,
    ):
    super().__init__(dataset_name=dataset_name, path_factory=path_factory)
    # other stuff here
```
and then you should also declare:
```
class ColumnBytesFileReader(abc.ABC):
    """A reader class for column bytes files to inherit upon for caches and cacheless"""

    def __init__(
        self,
        dataset_name: Optional[str] = None,
        path_factory: WickerPathFactory = S3PathFactory(),
    ) -> None:
        super().__init__()
        self._column_bytes_root_path = path_factory.get_column_concatenated_bytes_files_path(dataset_name=dataset_name)
        self._dataset_name = dataset_name
        self._path_factory = path_factory
```

Reply:
@convexquad do you prefer this even if the path factory is not used for anything but setting the column bytes root path?

Reply:
Regardless I made the changes so let me know what you think.

Reply:
@pickles-bread-and-butter I do think this is the best version of these classes since the column file bytes writer, reader and cache are all symmetric in that they all take a dataset_name and path_factory. Another refactoring that changes all three classes at once in a similar way to remove the path factory does indeed makes sense, but let's not do it now.